fabtests/ubertest/uber.c

```diff
@@ -328,6 +328,55 @@ static int ft_recv_test_info(void)
 	return 0;
 }
 
+static int ft_exchange_uint32(uint32_t local, uint32_t *remote)
+{
+	uint32_t local_net = htonl(local);
+	int ret;
+
+	ret = ft_sock_send(sock, &local_net, sizeof local);
+	if (ret) {
+		FT_PRINTERR("ft_sock_send", ret);
+		return ret;
+	}
+
+	ret = ft_sock_recv(sock, remote, sizeof *remote);
+	if (ret) {
+		FT_PRINTERR("ft_sock_recv", ret);
+		return ret;
+	}
+
+	*remote = ntohl(*remote);
+
+	return 0;
+}
+
+static int ft_skip_info(struct fi_info *hints, struct fi_info *info)
+{
+	uint32_t remote_protocol, skip, remote_skip;
+	size_t len;
+	int ret;
+
+	//check needed to skip utility providers, unless requested
+	skip = (!ft_util_name(hints->fabric_attr->prov_name, &len) &&
+		strcmp(hints->fabric_attr->prov_name,
+		info->fabric_attr->prov_name)) ? 1 : 0;
+
+	//make sure remote side is using the same protocol
+	ret = ft_exchange_uint32(info->ep_attr->protocol, &remote_protocol);
+	if (ret)
+		return ret;
+
+	if (info->ep_attr->protocol != remote_protocol)
+		return 1;
+
+	ret = ft_exchange_uint32(skip, &remote_skip);
+	if (ret)
+		return ret;
+
+	return skip || remote_skip ? 1 : 0;
```

Comment:
nit: can drop ? : and return boolean value directly
also extra blank line at end of function.